src/bitcoin/mod.rs

```diff
@@ -20,6 +21,27 @@ type SpentCoin = (bitcoin::OutPoint, bitcoin::Txid, i32, u32);
 
 const COINBASE_MATURITY: i32 = 100;
 
+// See https://github.com/bitcoin/bitcoin/blob/v27.1/src/kernel/chainparams.cpp.
+/// The expected genesis block hash.
+pub fn expected_genesis_hash(network: &bitcoin::Network) -> bitcoin::BlockHash {
```

Comment:
nit: I think those constants are already provided by rust-bitcoin https://docs.rs/bitcoin/0.32.2/bitcoin/constants/struct.ChainHash.html

Reply:
Ah, looks like they are in reversed byte order though..

Reply:
I've now removed this function and use the suggested constant instead.

For the Electrum genesis hash, I now just query the block hash with height 0 rather than the server features, which was providing the hash in reverse order. I may need to get the server features for other sanity checks at some point, e.g. protocol version, but it seemed natural to query the blockchain for the genesis hash.